src/rpc/core_rpc_server.cpp

```diff
@@ -1111,9 +1216,66 @@ namespace cryptonote
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::should_use_bootstrap_daemon()
+  {
+    if (m_bootstrap_daemon_address.empty())
+      return false;
+    auto current_time = std::chrono::system_clock::now();
+    if (current_time - m_bootstrap_height_check_time > std::chrono::seconds(30))  // update every 30s
+    {
+      m_bootstrap_height_check_time = current_time;
+
+      uint64_t top_height;
+      crypto::hash top_hash;
+      m_core.get_blockchain_top(top_height, top_hash);
+      ++top_height; // turn top block height into blockchain height
+      uint64_t target_height = m_core.get_target_blockchain_height();
+
+      m_should_use_bootstrap_daemon = top_height < target_height;
```

Comment:
I'd only do this if we're 100 blocks behind or suchlike, or you'll end up switching back and forth. Or maybe never switch it back on if it was switched off earlier. Or only switch is back on if the difference is way higher, like 5k blocks, say. Otherwise if this gets enabled by default, you end up not being able to trust your own daemon.

Reply:
Good point. I made a change so that it never goes back to the bootstrap mode once the sync is complete.

Reply:
There's a data race on `m_should_use_bootstrap_daemon`, doesn't look like a problem though (except possibly causing log spam).

Reply:
I'd say 3 months work, maybe 13k blocks. But at this point anything works